cli/src/cli.rs

```diff
@@ -45,13 +47,85 @@ pub(crate) async fn provision(sub: &ArgMatches) -> Result<()> {
     Ok(())
 }
 
-pub(crate) async fn ssh(sub: &ArgMatches) -> Result<()> {
-    let needle: String = sub.value_of_t("host")?;
+pub(crate) async fn ssh(sub: &ArgMatches, cluster: ClusterHandle) -> Result<()> {
     let mut args = sub.values_of_lossy("args").unwrap_or_default();
+    let job: Vec<String> = sub.values_of_t("job").unwrap_or_default();
 
-    let ip = find_instance(needle.as_str())
-        .await
-        .map_or_else(|| needle.clone(), |i| i.public_ip);
+    let namespace: String = sub
+        .value_of_t("namespace")
+        .unwrap_or(env::var("NOMAD_NAMESPACE")?);
```

Comment:
I'd replace most `unwrap`s in this PR with `expect` so we have some idea of wtf goes on when it fails...

Reply:
Absolutely agree. There are a lot of things I wanted to clean up here, and this was one of them. I'd also like to make the imports a bit more idiomatic, and see if I could optimize some of the find_needle functions, but I think it'd be better to merge and then add this later. But the error messages are important enough, that I can probably get to that over the weekend sometime :crossed_fingers:

Reply:
@manveru, how would you feel about simply checking the `IN_NIX_SHELL` var (and maybe some other BITTE_* var as well) at the beginning of main, and provide an error message if it doesn't exist advising that one must run bitte-cli from a bitte shell?